### PR TITLE
Sort multiselects alphabetically

### LINKED ISSUE #16317: Upsells order changing around.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [x] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

In a product's upsells admin area:
1. Add and save 3 upsells (A, B, C) to a product.
2. Remove B and C. Without saving, add C then B.
3. Observe order of upsells changes back to A, B, C. Expected order is A, C, B.

(This is most likely a bug in select2/selectWoo.)

## Expected/actual behavior

When I follow those steps, I see upsells in original order.

I was expecting to see upsells in new order.

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

cc @dougaitken

### PR DESCRIPTION
**Pull request by claudiulodro**
Fixes #16317.

It doesn't make any sense for a user to try and make a custom order in multi-selects (especially since the upsells order is random on the frontend).

This keeps selected elements in multi-selects ordered alphabetically by text.

To test: Follow same steps from #16317.

### CODE REVIEW

**Comment by mikejolley** [comment]
@claudiulodro Missing support for the sortable selects?

**Comment by claudiulodro** [comment]
@mikejolley I'm not sure I understand. Can you elaborate? Is that a feature WC uses? Should multiselects be drag-n-drop sortable?

**Comment by mikejolley** [comment]
It's something that can be enabled here: https://github.com/woocommerce/woocommerce/blob/master/assets/js/admin/wc-enhanced-select.js#L117

**Comment by claudiulodro** [comment]
I've moved the auto-sorting out of selectWoo and into wc-enhanced-select.js. 

I think having auto-sorting built into selectWoo will cause problems for people, since it caused problems for us. Since it's not a backwards-compatible feature I'm not going to include it in selectWoo at the moment.

File: assets/js/admin/wc-enhanced-select.js [hunk 1]
```diff
@@ -131,6 +131,24 @@ jQuery( function( $ ) {
...
+							var $children = $( this ).children();
+							$children.sort(function(a, b){
+ 								var atext = a.text.toLowerCase();
+       							var btext = b.text.toLowerCase();
+
+       							if (atext > btext) {
+         							return 1;
+       							}
+       							if (atext < btext) {
+         							return -1;
```
Comment (claudiulodro) [inline assets/js/admin/wc-enhanced-select.js]: This is some sort of rendering error in GitHub. [Indentation is correct](https://github.com/woocommerce/woocommerce/blob/d39f4bbe17778818961322e9f96ca22bf005bc0b/assets/js/admin/wc-enhanced-select.js#L142-L148)

**Pull request merged**